### PR TITLE
docs: Appium 1 EOL plan

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
## Proposed changes

I see a need for such document for both developers and users of Appium. It adds more transparency and visibility to what is being done and to what is expected to be done. I'm not quite sure where to put this document, it could even stay as a PR, although it is important to keep it publicly visible for the lifetime of this transition period.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: docs/en/about-appium/eol.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  appium-gulp-plugins (now `@appium/gulp-plugins`).
+- Added support of plugins
+- Added a CLI to manage drivers and plugins
+- Dropped support of JSONWP capabilities. Only W3C-compatible clients are now supported.
+- The list of supported command line arguments has been modified to exclude driver-specific ones.
+- Drivers compatible to Appium 2 must include a special manifest in their package.json
+
+## Transition Plan
+
+The preliminary date of Appium 2 release is *1st of January 2022*. After this date we won't
```
Comment (KazuCocoa) [inline docs/en/about-appium/eol.md:27]: @jlipps for this date?

Comment (jlipps) [reply docs/en/about-appium/eol.md:27]: It should be much sooner, like in the next few months

Comment (jlipps) [reply docs/en/about-appium/eol.md:27]: however i'm ok if we want a maintenance plan for 1.x that extends beyond 2.0 release a bit

**Review by jlipps — COMMENTED** [review]
this is awesome thanks @mykola-mokhnach! i like your suggestions, let's keep talking as a team about these things for a bit before merging/publicizing

File: docs/en/about-appium/eol.md (hunk 1 above)
Comment (KazuCocoa) [reply docs/en/about-appium/eol.md:27]: We can release 1.22.0 as the final version as 1.x, or 1.23.0?
(At least, we should release 1.22.0 with this 2.0 announcement on the release page to notify this 2.0 update widely I think.)

For now, it seems we can mark this line as tentative...? (and merge this PR to modify this good documentation more?)

**Review by KazuCocoa — APPROVED** [review]

File: docs/en/about-appium/eol.md (hunk 1 above)
Comment (jlipps) [reply docs/en/about-appium/eol.md:27]: I think we should say preliminary release date is like Aug 1 2021. and after this date we won't add features to 1.x. but will continue to provide urgent bugfixes to 1.x through Dec 31 2021.

Comment (boneskull) [reply docs/en/about-appium/eol.md:27]: let's not commit to any dates until we can commit to dates 😄

Comment (KazuCocoa) [reply docs/en/about-appium/eol.md:27]: seems good for me👌

**Comment by jlipps** [comment]
closed in favor of #15711

**Pull request closed without merging**